### PR TITLE
chore: WrapSds from family_utils.h

### PR DESCRIPTION
**Pull request by dranikpg**
Remove some of the c'stlye code

### CODE REVIEW

File: src/server/stream_family.cc [hunk 1]
```diff
@@ -1306,23 +1298,18 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
...
 
-    return OpStatus::OK;
+  // Awake readers blocked on this group
+  auto blocking_controller = op_args.db_cntx.ns->GetBlockingController(op_args.shard->shard_id());
+  if (blocking_controller) {
+    blocking_controller->AwakeWatched(op_args.db_cntx.db_index, key);
   }
 
-  return OpStatus::SKIPPED;
+  return OpStatus::OK;
```
Comment (romange) [inline src/server/stream_family.cc:1312]: I understand the logic has been changed here. Is it something we cover with tests?

Comment (dranikpg) [reply src/server/stream_family.cc:1312]: Reply to last 3 comments: FindGroup now return SKIPPED if the group is nullptr

So 
```c++
auto res = FindGroup();
if (!res)
  return res.status();
if (!res->cg)
  return res.status();
```

which appears multiple times turns into

```c++
auto res = FindGroup();
RETURN_ON_BAD_STATUS(res);
```

File: src/server/stream_family.cc [hunk 2]
```diff
@@ -1262,12 +1256,10 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
...
 
       // Try to get the consumer. If not found, create a new one.
-      op_args.shard->tmp_str1 =
-          sdscpylen(op_args.shard->tmp_str1, opts.consumer.data(), opts.consumer.size());
-      if ((consumer = streamLookupConsumer(cgr_res->cg, op_args.shard->tmp_str1, SLC_NO_REFRESH)) ==
-          nullptr) {
-        consumer = streamCreateConsumer(cgr_res->cg, op_args.shard->tmp_str1, nullptr, 0,
-                                        SCC_NO_NOTIFY | SCC_NO_DIRTIFY);
+      auto cname = WrapSds(opts.consumer);
+      if ((consumer = streamLookupConsumer(cgr_res->cg, cname, SLC_NO_REFRESH)) == nullptr) {
```
Comment (romange) [inline src/server/stream_family.cc:1260]: nit: i personally do not like these double level expressions.

File: src/server/stream_family.cc [hunk 3]
```diff
@@ -1306,23 +1298,18 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
...
-    raxRemove(cgr_res->s->cgroups, (uint8_t*)(gname.data()), gname.size(), NULL);
-    streamFreeCG(cgr_res->cg);
+  RETURN_ON_BAD_STATUS(cgr_res);
 
-    // Awake readers blocked on this group
-    auto blocking_controller = op_args.db_cntx.ns->GetBlockingController(op_args.shard->shard_id());
-    if (blocking_controller) {
-      blocking_controller->AwakeWatched(op_args.db_cntx.db_index, key);
-    }
+  raxRemove(cgr_res->s->cgroups, (uint8_t*)(gname.data()), gname.size(), NULL);
```
[the remark this answers is not in the document]
Comment (romange) [reply src/server/stream_family.cc:1303]: ping here? Did you validate that based on the code `cgr_res->cg` is always defined here?
Please add DCHECK before raxRemove

[the remark this answers is not in the document]
Comment (romange) [reply src/server/stream_family.cc:1303]: ah I see now why, nm

**Review by romange — DISMISSED** [review]
👏🏼 👏🏼 👏🏼  Very Nice!

File: src/server/family_utils.cc [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#include "server/family_utils.h"
+
+#include "base/logging.h"
+
+namespace dfly {
+
+sds WrapSds(std::string_view s) {
+  static __thread sds tmp_sds = nullptr;
+  if (tmp_sds == nullptr)
+    tmp_sds = sdsempty();
```
Comment (kostasrim) [inline src/server/family_utils.cc:10]:
```suggestion
  static __thread sds tmp_sds = sdsempry();
```

Comment (kostasrim) [reply src/server/family_utils.cc:10]: @dranikpg I did not approve since Roman did. If you apply this change just let me know and will accept quickly

**Comment by romange** [comment]
@kostasrim  please fix your typo

File: src/server/family_utils.cc [hunk 5]
```diff
@@ -0,0 +1,43 @@
+#include "server/family_utils.h"
+
+#include "base/logging.h"
+
+namespace dfly {
+
+sds WrapSds(std::string_view s) {
+  static __thread sds tmp_sds = sdsempry();
```
Comment (kostasrim) [inline src/server/family_utils.cc:8]:
```suggestion
  static thread_local sds tmp_sds = sdsempty();
```

**Review by kostasrim — APPROVED** [review]

File: src/server/stream_family.cc (hunk 2 above)
Comment (dranikpg) [reply src/server/stream_family.cc:1260]: Not me who introduced it ☝🏻

**Pull request merged**